### ISSUE TITLE
Bug found in prefer-const

### ISSUE THREAD

**Issue by ffxsam** [eslint/eslint#5473]
Version 2.2.0

Given this code:

```js
  let someConst;

  someConst = wow();
```

ESlint tells me that `someConst` should be a `const` instead of a `let` because it's never modified.

**Comment by platinumazure** [comment]
From your example, as far as I can tell, `someConst` is in fact never modified. You can write the example as follows:

```js
const someConst = wow();
```

What about this is a bug, exactly?

**Comment by ffxsam** [comment]
Oh shoot, I see the problem. It thinks it's pointless to declare it without defining it, and then define it below, so I should use a const. That makes sense.. *unless* of course you're religious about hoisting variables, which I am:

```
    let setLater, result;
    
    somePreparation();
    // More code
    result = doThings();
    setLater.value = result;
```

ESlint complains that `result` should be a const. But let's say there's a bunch of code that has to be executed before I can set `result`. See the dilemma?

**Comment by platinumazure** [comment]
Sounds like you might not want to use `prefer-const` in that case. I would suggest disabling it in config (or inline in your file, if it's a rare enough thing).

You could also introduce new blocks, which allows you to feel good about hoisting while also using `const`.

In any case, I don't think there's really a bug here. Sometimes not all ESLint rules are a good fit for a user's use case.

**Comment by ffxsam** [comment]
Yeah, after discussing it, I realize it's not really a bug. Thanks for providing some clarity!
